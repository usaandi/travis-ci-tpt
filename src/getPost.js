const axios = require('axios');

module.exports = function getPost(postId, callback) {
  axios.get(`https://jsonplaceholder.typicode.com/posts/${postId}`)
    .then((response) => {
      callback(response.data);
    })
    .catch((error) => {
      console.log(error);
    });
};
