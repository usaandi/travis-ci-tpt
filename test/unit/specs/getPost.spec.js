import getPost from '../../../src/getPost';

describe('getPost', () => {
  it('get post 5', (done) => {
    getPost(5, (data) => {
      console.log('post5', data);
      expect(data).toMatchSnapshot();
      done();
    });
  });

  it('get post 7', (done) => {
    getPost(7, (data) => {
      console.log('post7', data);
      expect(data).toMatchSnapshot();
      done();
    });
  });
});
