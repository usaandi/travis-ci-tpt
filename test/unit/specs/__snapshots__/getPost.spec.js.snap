// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPost get post 5 1`] = `
Object {
  "body": "repudiandae veniam quaerat sunt sed
alias aut fugiat sit autem sed est
voluptatem omnis possimus esse voluptatibus quis
est aut tenetur dolor neque",
  "id": 5,
  "title": "nesciunt quas odio",
  "userId": 1,
}
`;

exports[`getPost get post 7 1`] = `
Object {
  "body": "dolore placeat quibusdam ea quo vitae
magni quis enim qui quis quo nemo aut saepe
quidem repellat excepturi ut quia
sunt ut sequi eos ea sed quas",
  "id": 7,
  "title": "magnam facilis autem",
  "userId": 1,
}
`;
